src/rcms/fm/app/level1/HCALxmlHandler.java

```diff
@@ -223,6 +223,10 @@ public String stripExecXML(String execXMLstring, ParameterSet<FunctionManagerPar
       for (StringT maskedApp: maskedAppArray) {
         //logger.info("[JohnLogVector] " + functionManager.FMname + ": about to start masking " + maskedApp.getString());
         String[] maskedAppParts = maskedApp.getString().split("_");
+        if (maskedAppParts.length != 2 || HCALlevelOneEventHandler.stringToInt(maskedAppParts[0]) || !(HCALlevelOneEventHandler.stringToInt(maskedAppParts[1]))){
```

Comment:
`StripExecXml()` should take input of all maskedapps only, so it's a mistake to mix `maskedFM` into its input.
I think the only way to validate input against all maskedapps is what you have done in the masker, i.e. to retrieve the lightGroup from DB, this seems a bit over kill to me. 
For me, it's good enough to make sure the split does not break the code and dis-entangle `maskedFM` and `maskedapps` in another issue.